Mathlib/Combinatorics/Additive/PluenneckeRuzsa.lean

```diff
@@ -141,80 +183,80 @@ theorem ruzsa_triangle_inequality_mul_mul_mul (A B C : Finset α) :
   rw [← div_inv_eq_mul, div_eq_mul_inv]
   exact ruzsa_triangle_inequality_mul_div_div _ _ _
 
 -- Auxiliary lemma towards the Plünnecke-Ruzsa inequality
 @[to_additive]
-private lemma card_mul_pow_le (hAB : ∀ A' ⊆ A, (A * B).card * A'.card ≤ (A' * B).card * A.card)
-    (n : ℕ) : (A * B ^ n).card ≤ ((A * B).card / A.card : ℚ≥0) ^ n * A.card := by
+private lemma card_mul_pow_le (hAB : ∀ A' ⊆ A, #(A * B) * #A' ≤ #(A' * B) * #A) (n : ℕ) :
+    #(A * B ^ n) ≤ (#(A * B) / #A : ℚ≥0) ^ n * #A := by
   obtain rfl | hA := A.eq_empty_or_nonempty
   · simp
   induction' n with n ih
   · simp
-  rw [_root_.pow_succ', ← mul_assoc, _root_.pow_succ', @mul_assoc ℚ≥0, ← mul_div_right_comm,
-    le_div_iff₀, ← cast_mul]
-  swap
-  · exact cast_pos.2 hA.card_pos
-  refine (Nat.cast_le.2 <| pluennecke_petridis_inequality_mul _ hAB).trans ?_
-  rw [cast_mul]
-  gcongr
+  refine le_of_mul_le_mul_right ?_ (by positivity : (0 : ℚ≥0) < #A)
+  calc
+    ((#(A * B ^ (n + 1))) * #A : ℚ≥0)
+      = #(A * B * B ^ n) * #A := by rw [_root_.pow_succ', ← mul_assoc]
+    _ ≤ #(A * B) * #(A * B ^ n) := mod_cast pluennecke_petridis_inequality_mul _ hAB
+    _ ≤ #(A * B) * ((#(A * B) / #A) ^ n * #A) := by gcongr
+    _ = (#(A * B) / #A) ^ (n + 1) * #A * #A := by field_simp; ring
 
 /-- The **Plünnecke-Ruzsa inequality**. Multiplication version. Note that this is genuinely harder
 than the division version because we cannot use a double counting argument. -/
 @[to_additive "The **Plünnecke-Ruzsa inequality**. Addition version. Note that this is genuinely
 harder than the subtraction version because we cannot use a double counting argument."]
-theorem pluennecke_ruzsa_inequality_pow_div_pow_mul (hA : A.Nonempty) (B : Finset α) (m n : ℕ) :
-    ((B ^ m / B ^ n).card) ≤ ((A * B).card / A.card : ℚ≥0) ^ (m + n) * A.card := by
+theorem pluennecke_ruzsa_inequality_pow_div_pow_mul (hA : A.Nonempty) (B : Finset G) (m n : ℕ) :
+    #(B ^ m / B ^ n) ≤ (#(A * B) / #A : ℚ≥0) ^ (m + n) * #A := by
   have hA' : A ∈ A.powerset.erase ∅ := mem_erase_of_ne_of_mem hA.ne_empty (mem_powerset_self _)
-  obtain ⟨C, hC, hCA⟩ :=
-    exists_min_image (A.powerset.erase ∅) (fun C ↦ (C * B).card / C.card : _ → ℚ≥0) ⟨A, hA'⟩
+  obtain ⟨C, hC, hCmin⟩ :=
+    exists_min_image (A.powerset.erase ∅) (fun C ↦ #(C * B) / #C : _ → ℚ≥0) ⟨A, hA'⟩
   rw [mem_erase, mem_powerset, ← nonempty_iff_ne_empty] at hC
-  refine (_root_.mul_le_mul_right <| cast_pos.2 hC.1.card_pos).1 ?_
-  norm_cast
-  refine (Nat.cast_le.2 <| ruzsa_triangle_inequality_div_mul_mul _ _ _).trans ?_
-  push_cast
-  rw [mul_comm _ C]
-  refine (mul_le_mul (card_mul_pow_le (mul_aux hC.1 hC.2 hCA) _)
-    (card_mul_pow_le (mul_aux hC.1 hC.2 hCA) _) (zero_le _) (zero_le _)).trans ?_
-  rw [mul_mul_mul_comm, ← pow_add, ← mul_assoc]
-  gcongr ((?_ ^ _) * Nat.cast ?_) * _
-  · exact hCA _ hA'
-  · exact card_le_card hC.2
+  obtain ⟨hC, hCA⟩ := hC
+  refine le_of_mul_le_mul_right ?_ (by positivity : (0 : ℚ≥0) < #C)
```

Comment:
hmm this makes me think `le_of_mul_le_mul_right` wants an explicit argument

Reply:
Possible, but out of scope ;)